### PR TITLE
refactor: rename addHunger to setHunger

### PR DESCRIPTION
**Pull request by artur-michalak**
## Description

Many users have reported errors related to consumables. The problem lies in the synchronization of the statebag with the player's metadata in qbx_core. However, I have made a few improvements in smallresurces that may be of value.

The name `addHunger` is confusing when replacing an old value instead of adding a new one to it.
A new function `addHunger` has appeared in PR, whose logic is more in line with its name.

Moreover, a hotfix has been placed in the code for https://github.com/Qbox-project/qbx_core/issues/457.

Debug commands at https://github.com/Qbox-project/qbx_adminmenu/pull/57

- [X] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [X] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: qbx_consumables/server.lua [hunk 1]
```diff
@@ -32,49 +64,40 @@ end
...
+        local sustenance = math.random(params.min, params.max)
         relieveStress(source, params.stressRelief.min, params.stressRelief.max)
-        addThirst(sustenance)
+
+        addThirst(source, sustenance)
     end)
 end
 
 for drink, params in pairs(config.consumables.drink) do
     exports.qbx_core:CreateUseableItem(drink, function(source, item)
```
Comment (artur-michalak) [inline qbx_consumables/server.lua:86]: I'm not sure if this is ever executed

Comment (TonybynMp4) [reply qbx_consumables/server.lua:86]: Wdym? It's executed when the resource starts

File: qbx_consumables/server.lua [hunk 2]
```diff
@@ -120,6 +143,54 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
...
+
+RegisterNetEvent('consumables:server:setThirst', function (amount)
+    setThirst(source, amount)
+end)
+
+RegisterNetEvent('consumables:server:setHunger', function (amount)
+    setHunger(source, amount)
+end)
+
+lib.addCommand('thrist', {
```
Comment (TonybynMp4) [inline qbx_consumables/server.lua]: Typo

File: qbx_consumables/server.lua [hunk 3]
```diff
@@ -120,6 +143,54 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
...
+        {
+            name = 'target',
+            type = 'playerId',
+            optional = true
+        }
+    },
+    restricted = 'group.admin'
+}, function (source, args, raw)
+    setHunger(args.target or source, args.amount)
+end)
```
Comment (TonybynMp4) [inline qbx_consumables/server.lua]: Those commands should, if they're even needed, be in qbx_adminmenu

File: qbx_consumables/server.lua (hunk 1 above)
Comment (artur-michalak) [reply qbx_consumables/server.lua:86]: Yes, but water and burger don't use it

Comment (artur-michalak) [reply qbx_consumables/server.lua:86]: The script works well with snacks from qbx_smallresources, but there are two places where snacks are configured, this is a problem

**Review by deadfire24 — APPROVED** [review]

**Review by TonybynMp4 — CHANGES_REQUESTED** [review]

**Comment by TonybynMp4** [comment]
IMO this pr shouldn't be.
You're saying it's "confusing" that the function is called addHunger, when it literally ***adds*** hunger..

File: qbx_consumables/server.lua [hunk 4]
```diff
@@ -32,49 +64,40 @@ end
...
-        local player = exports.qbx_core:GetPlayer(source)
-        if not player then return end
-
         local drank = lib.callback.await('consumables:client:DrinkAlcohol', source, params.alcoholLevel, params.anim, params.prop)
         if not drank then return end
         if not exports.ox_inventory:RemoveItem(source, item.name, 1, nil, item.slot) then return end
-        local playerState = Player(source).state
 
-        local sustenance = (playerState.thirst or 0) + math.random(params.min, params.max)
+        local sustenance = math.random(params.min, params.max)
```
Comment (TonybynMp4) [inline qbx_consumables/server.lua:78]: Why this change? you're literally allowing an error to occur..

File: qbx_consumables/server.lua [hunk 5]
```diff
@@ -120,6 +143,54 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
...
+    setHunger(source, amount)
+end)
+
+RegisterNetEvent('consumables:server:setThirst', function (amount)
+    setThirst(source, amount)
+end)
+
+RegisterNetEvent('consumables:server:setHunger', function (amount)
+    setHunger(source, amount)
+end)
```
Comment (TonybynMp4) [inline qbx_consumables/server.lua]: The goal isn't to set a new value, it's to add to the current value

File: qbx_consumables/server.lua (hunk 4 above)
Comment (artur-michalak) [reply qbx_consumables/server.lua:78]: check the implementation of addThirst

File: qbx_consumables/server.lua (hunk 5 above)
Comment (artur-michalak) [reply qbx_consumables/server.lua]: Function should describe what it does, not what it gets. qbx_core gives the full value, which replaces the old thirst or hunger level.
That's what line
`local sustenance = (playerState.hunger or 0) + math.random(params.min, params.max)`
does. AddSomething is linked to the old value, SetSomething sets the new value with no relation to the old value.

File: qbx_consumables/server.lua (hunk 3 above)
Comment (artur-michalak) [reply qbx_consumables/server.lua]: yes you are right

File: qbx_consumables/server.lua [hunk 6]
```diff
@@ -120,6 +143,20 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
...
+RegisterNetEvent('consumables:server:addThirst', function (amount)
+    setThirst(source, amount)
+end)
+--- @deprecated confusing name
+RegisterNetEvent('consumables:server:addHunger', function (amount)
+    setHunger(source, amount)
+end)
+
+RegisterNetEvent('consumables:server:setThirst', function (amount)
+    setThirst(source, amount)
```
Comment (ArturMichalak) [inline qbx_consumables/server.lua:157]:
```suggestion
    if not GetInvokingResource() then return end
    setThirst(source, amount)
```

File: qbx_consumables/server.lua [hunk 7]
```diff
@@ -120,6 +143,20 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
...
-RegisterNetEvent('consumables:server:addThirst', addThirst)
 
-RegisterNetEvent('consumables:server:addHunger', addHunger)
+--- @deprecated confusing name
+RegisterNetEvent('consumables:server:addThirst', function (amount)
+    setThirst(source, amount)
+end)
+--- @deprecated confusing name
+RegisterNetEvent('consumables:server:addHunger', function (amount)
+    setHunger(source, amount)
```
Comment (ArturMichalak) [inline qbx_consumables/server.lua:153]:
```suggestion
    if not GetInvokingResource() then return end
    setHunger(source, amount)
```

File: qbx_consumables/server.lua [hunk 8]
```diff
@@ -120,6 +143,20 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
 end)
 
 -- Added for ox_inv until I make a proper qbx bridge
-RegisterNetEvent('consumables:server:addThirst', addThirst)
 
-RegisterNetEvent('consumables:server:addHunger', addHunger)
+--- @deprecated confusing name
+RegisterNetEvent('consumables:server:addThirst', function (amount)
+    setThirst(source, amount)
```
Comment (ArturMichalak) [inline qbx_consumables/server.lua:149]:
```suggestion
    if not GetInvokingResource() then return end
    setThirst(source, amount)
```

File: qbx_consumables/server.lua [hunk 9]
```diff
@@ -120,6 +143,20 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
...
+RegisterNetEvent('consumables:server:addHunger', function (amount)
+    setHunger(source, amount)
+end)
+
+RegisterNetEvent('consumables:server:setThirst', function (amount)
+    setThirst(source, amount)
+end)
+
+RegisterNetEvent('consumables:server:setHunger', function (amount)
+    setHunger(source, amount)
```
Comment (ArturMichalak) [inline qbx_consumables/server.lua:161]:
```suggestion
    if not GetInvokingResource() then return end
    setHunger(source, amount)
```

File: qbx_consumables/server.lua (hunk 8 above)
Comment (ArturMichalak) [reply qbx_consumables/server.lua:149]: I thought about adding such a line, however, ox_inventory qb bridge could break down.

**Comment by Manason** [comment]
> IMO this pr shouldn't be. You're saying it's "confusing" that the function is called addHunger, when it literally _**adds**_ hunger..

addHunger should increment current hunger by some amount.
setHunger would replace current hunger with some amount.

Didn't read the PR, just want to clarify on what names should go with what in case there is confusion.

File: qbx_consumables/server.lua [hunk 10]
```diff
@@ -120,6 +143,54 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
 end)
 
 -- Added for ox_inv until I make a proper qbx bridge
-RegisterNetEvent('consumables:server:addThirst', addThirst)
 
-RegisterNetEvent('consumables:server:addHunger', addHunger)
+--- @deprecated confusing name
```
Comment (xhorntail) [inline qbx_consumables/server.lua]: I think the deprecation definition should direct the user to the new implementation, rather than the reason for deprecation

File: qbx_consumables/server.lua [hunk 11]
```diff
@@ -1,17 +1,49 @@
 local config = require 'qbx_consumables.config'
 
-local function addHunger(amount)
+--- sets player nutrition level in statebag
```
Comment (xhorntail) [inline qbx_consumables/server.lua]: This is setting the state for `hunger`, probably best to use that nomenclature rather than `nutrition` in the docstring

File: qbx_consumables/server.lua [hunk 12]
```diff
@@ -1,17 +1,49 @@
...
+
+    player.Functions.SetMetaData('hunger', amount)
+    --- hotfix
+end
+
+--- raises player nutrition level in statebag
+--- @param source number
+--- @param amount number
+local function addHunger(source, amount)
+    local hunger = Player(source).state.hunger or 0
```
Comment (artur-michalak) [inline qbx_consumables/server.lua:30]: How about “assert” when state.hunger = nil?

File: qbx_consumables/server.lua (hunk 10 above)
Comment (artur-michalak) [reply qbx_consumables/server.lua]: Is currently good?

File: qbx_consumables/server.lua (hunk 12 above)
Comment (xhorntail) [reply qbx_consumables/server.lua:30]: Hmm, I have a couple concerns that I'm reviewing locally, give me a few and I'll get back to ya.
In the meantime, @Manason am I to assume by [qbx_core issue #456](https://github.com/Qbox-project/qbx_core/issues/456) that only OneSync Infinity is under scope of support?

File: qbx_consumables/server.lua [hunk 13]
```diff
@@ -120,6 +144,34 @@ lib.callback.register('consumables:server:usedItem', function(source, item)
...
+---client-side call, sets player thirst level
+---@param amount number
+RegisterNetEvent('consumables:server:setThirst', function (amount)
+    if not GetInvokingResource() then setThirst(source, amount) end
+end)
+
+---client-side call, sets player hunger level
+---@param amount number
+RegisterNetEvent('consumables:server:setHunger', function (amount)
+    if not GetInvokingResource() then setHunger(source, amount) end
```
Comment (xhorntail) [inline qbx_consumables/server.lua:171]: Can you explain the purpose behind the `not GetInvokingResource()` conditional here for me?

File: qbx_consumables/server.lua (hunk 12 above)
Comment (xhorntail) [reply qbx_consumables/server.lua:28]: @ArturMichalak I've sent a PR to your fork that you can review.

**Review by xhorntail — APPROVED** [review]
Sorry, it's an early morning for me, posted in the wrong spot initially.
Anyhow, @ArturMichalak  I've sent a PR to your fork that you can review/implement as you see fit. I'm dropping out of this for now.

**Comment by artur-michalak** [comment]
Thanks a lot, I didn't know how to describe these functions well.
I'll adapt it soon.

File: qbx_consumables/server.lua (hunk 12 above)
Comment (BerkieBb) [reply qbx_consumables/server.lua:30]: We only support OneSync Infinity yeah

**Review by BerkieBb — APPROVED** [review]

**Review by Manason — APPROVED** [review]
skimmed
